Xamarin.Essentials/Share/Share.shared.cs

```diff
@@ -31,13 +33,33 @@ public static Task RequestAsync(ShareFileRequest request)
                 throw new ArgumentNullException(nameof(request));
 
             if (request.File == null)
-                throw new ArgumentException($"The {nameof(request.File)} in the request is invalid.");
+                throw new ArgumentException(FileNullExeption(nameof(request.File)));
 
             if (string.IsNullOrEmpty(request.File.FullPath))
-                throw new ArgumentException($"The request file's {nameof(request.File.FullPath)} is invalid.");
+                throw new ArgumentException(EmptyPathExeption(nameof(request.File.FullPath)));
+
+            return PlatformRequestAsync((ShareMultipleFilesRequest)request);
+        }
+
+        public static Task RequestAsync(ShareMultipleFilesRequest request)
+        {
+            if (request == null)
+                throw new ArgumentNullException(nameof(request));
+
+            if (!(request.Files?.Count() > 0))
+                throw new ArgumentException(FileNullExeption(nameof(request.Files)));
+
+            if (request.Files.Any(file => string.IsNullOrEmpty(file.FullPath)))
+                throw new ArgumentException(EmptyPathExeption(nameof(ShareFile.FullPath)));
 
             return PlatformRequestAsync(request);
         }
+
+        static string FileNullExeption(string file)
+            => $"The request file's {file} is invalid.";
```

Comment:
Should this be constructing new exception objects? Then we don't need to worry about type. Or even should this be actually throwing? `EnsureNotEmptyPath` or something? Then everyone can just call it and not worry about anything at all.

Reply:
This is the old logic, we tried not to change anything

Reply:
Let's go with it for now. I have no real preference and the exception police/gurus can tell us where we went wrong. Anyone a guru out there?

Reply:
We must leave an exception so that the developer understands that the files cannot be sent
And handle the exception in the app

Reply:
Ah, I was meaning leave it as you had it. So I think we agree 😊